lib/node_modules/@stdlib/math/base/special/sincos/lib/kernel_rem_pio2.js

```diff
@@ -0,0 +1,379 @@
+		if ( z >= TWO24 ) {
+			fw = (TWON24*z)|0;
+			IQ[ jz ] = ( z - (TWO24*fw) )|0;
+			jz += 1;
+			q0 += 24;
+			IQ[ jz ] = fw;
+		} else {
+			IQ[ jz ] = z|0;
+		}
+	}
+	// Convert integer "bit" chunk to floating-point value...
+	fw = ldexp( 1.0, q0 );
+	for ( i = jz; i >= 0; i-- ) {
+		q[ i ] = fw * IQ[i];
+		fw *= TWON24;
+	}
+	// Compute `PIO2[0,...,jp]*q[jz,...,0]`...
+	for ( i = jz; i >= 0; i-- ) {
+		fw = 0.0;
+		for ( k = 0; k <= jp && k <= jz-i; k++ ) {
+			fw += PIO2[ k ] * q[ i+k ];
+		}
+		FQ[ jz-i ] = fw;
+	}
+	// Compress `FQ[]` into `y[]`...
+	switch ( prec ) {
+	case 0:
+		fw = 0.0;
+		for ( i = jz; i >= 0; i-- ) {
+			fw += FQ[ i ];
+		}
+		if ( ih === 0 ) {
+			y[ 0 ] = fw;
+		} else {
+			y[ 0 ] = -fw;
+		}
+		break;
+	case 1:
+	case 2:
+		fw = 0.0;
+		for ( i = jz; i >= 0; i-- ) {
+			fw += FQ[ i ];
+		}
+		if ( ih === 0 ) {
+			y[ 0 ] = fw;
+		} else {
+			y[ 0 ] = -fw;
+		}
+		fw = FQ[ 0 ] - fw;
+		for ( i = 1; i <= jz; i++ ) {
+			fw += FQ[i];
+		}
+		if ( ih === 0 ) {
+			y[ 1 ] = fw;
+		} else {
+			y[ 1 ] = -fw;
+		}
+		break;
+	case 3:
+		for ( i = jz; i > 0; i-- ) {
+			fw = FQ[ i-1 ] + FQ[ i ];
+			FQ[ i ] += FQ[ i-1 ] - fw;
+			FQ[ i-1 ] = fw;
+		}
+		for ( i = jz; i > 1; i-- ) {
+			fw = FQ[ i-1 ] + FQ[ i ];
+			FQ[ i ] += FQ[ i-1 ] - fw;
+			FQ[ i-1 ] = fw;
+		}
+		fw = 0.0;
+		for ( i = jz; i >= 2; i-- ) {
+			fw += FQ[ i ];
+		}
+		if ( ih === 0 ) {
+			y[ 0 ] = FQ[ 0 ];
+			y[ 1 ] = FQ[ 1 ];
+			y[ 2 ] = fw;
+		} else {
+			y[ 0 ] = -FQ[ 0 ];
+			y[ 1 ] = -FQ[ 1 ];
+			y[ 2 ] = -fw;
+		}
+		break;
+	default:
+		break;
+	}
+	return ( n & 7 );
+} // end FUNCTION compute()
+
+
+// MAIN //
+
+/**
+* Returns the last three digits of `N` with `y = x - N*pi/2` so that `|y| < pi/2`.
+*
+* ## Method
+*
+* The method is to compute the integer (mod 8) and fraction parts of `(2/pi)*x` without doing the full multiplication. In general, we skip the part of the product that is known to be a huge integer (more accurately, equals 0 mod 8 ). Thus, the number of operations are independent of the exponent of the input.
+*
+* @private
+* @param {PositiveNumber} x - input value
+* @param {Array<number>} y - output result in an array of double precision numbers
+* @param {PositiveInteger} e0 - the exponent of `x[0]` (must be <= 16360)
+* @param {PositiveInteger} nx - dimension of `x[]`
+* @param {PositiveInteger} prec - precision in bits (can be 24 (single), 53 (double), 64 (extended), 113 (quad))
+* @returns {number} last three digits of `N`
+*/
+function kernelRemPio2( x, y, e0, nx, prec ) {
+	var fw;
+	var jk;
+	var jv;
+	var jx;
+	var jz;
+	var q0;
+	var i;
+	var j;
+	var m;
+
+	// Initialize `jk`:
+	jk = INIT_JK[ prec ];
+
+	// Determine `jx`, `jv`, `q0` (note that `q0 < 3`):
+	jx = nx - 1;
+	jv = ( (e0 - 3) / 24 )|0;
+	if ( jv < 0 ) {
+		jv = 0;
+	}
+	q0 = e0 - (24 * (jv + 1));
+
+	// Set up `F[0]` to `F[jx+jk]` where `F[jx+jk] = IPIO2[jv+jk]`:
+	j = jv - jx;
+	m = jx + jk;
+	for ( i = 0; i <= m; i++, j++ ) {
```

Comment:
Disable the lint rule on this line.

// eslint-disable-line no-plusplus

Can then remove top-level disabling of lint rule.